### PR TITLE
V0.12

### PR DESCRIPTION
**Pull request by kaiquelupo**
The version 0.0.12 has the following changes: 

- Upgrade Pulumi Version to ^3.0.0
- Update how Flex Plugins are deployed and release. Now we are using the Flex Plugin CLI. The documentation on how to use it and new attributes is coming with pull request #3 . 
- Fix major bug: replace Twilio client direct instantiation with proper method to handle login using Auth Token and API Key. 
- Fix minor bug: when deploying serverless or flex plugins, the hashes of the folders (to check in a next deploy if code has changed) were being saved with a version before deployment. Depending on the deploy, it can change code as well. Therefore, now the hash is verified in the beginning and in the end of the deploy. 
- Fix major bug: the usage of dotenv were returning a dependency error. To solve that, the package is being required inside the method that uses it. (I couldn't find another way of solving it. I think it is related to how Pulumi CLI loads code)

### CODE REVIEW

**Review by vernig — APPROVED** [review]
Fix #4

File: src/utils/index.ts
Comment (vernig) [inline src/utils/index.ts:12]: I would move the `require()` on the top of the file (nice to have)

**Comment by vernig** [comment]
Generally LGTM @kaiquelupo, but I don't have a deep knowledge of the provider to be providing an insightful review. 

Just one thing. I ran this version of the provider on the outbound-messaging example (the fixed one) and I get the following error that I didn't get with the previous version of the provider: 

```
View Live: https://app.pulumi.com/gverni/outbound-messaging/dev3/updates/1

     Type                               Name                      Status                  Info
 +   pulumi:pulumi:Stack                outbound-messaging-dev3   **creating failed**     1 error; 2 messages
 +   ├─ pulumi-nodejs:dynamic:Resource  outbound-sms-flex-plugin  **creating failed**     1 error
 +   └─ pulumi-nodejs:dynamic:Resource  flex-workspace            **creating failed**     1 error
 
Diagnostics:
  pulumi:pulumi:Stack (outbound-messaging-dev3):
    The plugin plugin-outbound-sms version (v3) is not compatible with this CLI command.
 
    Run $ twilio flex:plugins:upgrade-plugin --install to upgrade your plugin.
 
    error: update failed
 
  pulumi-nodejs:dynamic:Resource (outbound-sms-flex-plugin):
    error: Error: Command failed: twilio flex:plugins:deploy --changelog=deployed by infra as code
``` 

I haven't had the time to investigate that properly, but can we hold off the merging until we make sure we can at least deploy that example?

File: src/utils/index.ts
Comment (kaiquelupo) [reply src/utils/index.ts:12]: Yep, it is related to that. Not sure what is happening. I've tried to install other libraries but they fail with the same error. I think it is related to how Pulumi runs the code.

**Comment by kaiquelupo** [comment]
> LGTM

Good point @vernig . This is happening because the outbound-messaging main branch is using the old Flex Plugin framework and this provider is now using the new Flex Plugin API. I have the outbound-messaging code updated and I will create the pull request there right now. After we merge there, we can merge here.

**Pull request merged**
